x/rollup/keeper/msg_server.go

```diff
@@ -130,7 +157,37 @@ func (k *Keeper) MintETH(ctx *sdk.Context, addr sdk.AccAddress, amount sdkmath.I
 			types.EventTypeMintETH,
 			sdk.NewAttribute(types.AttributeKeyL1DepositTxType, types.L1UserDepositTxType),
 			sdk.NewAttribute(types.AttributeKeyToCosmosAddress, addr.String()),
-			sdk.NewAttribute(types.AttributeKeyAmount, hexutil.Encode((amount.BigInt().Bytes()))),
+			sdk.NewAttribute(types.AttributeKeyAmount, hexutil.Encode(amount.BigInt().Bytes())),
+		),
+	})
+	return nil
+}
+
+// BurnETH burns ETH from an account where the amount is in wei
+func (k *Keeper) BurnETH(ctx sdk.Context, addr sdk.AccAddress, amount sdkmath.Int) error { //nolint:gocritic // hugeParam
+	coins := sdk.NewCoins(sdk.NewCoin(types.ETH, amount))
+
+	// Transfer the coins to withdraw from the user account to the mint module
+	err := k.bankkeeper.SendCoinsFromAccountToModule(ctx, addr, types.MintModule, coins)
+	if err != nil {
+		return fmt.Errorf("failed to send withdrawal coins from user account %v to mint module: %v", addr, err)
+	}
+
+	// Burn the ETH coins from the mint module
+	if err := k.bankkeeper.BurnCoins(ctx, types.MintModule, coins); err != nil {
```

Comment:
I chose to burn the coins from the mint module, however they could also be burned (or even minted for deposits) through the bank module too. Lmk if you have any preference which module we go through

Reply:
Bank keeper is probably better. Mint module [appears to be](https://docs.cosmos.network/main/build/modules/mint) mainly for calculating staking rewards and token inflation, which is what Monomer helps avoid.

Reply:
Sounds good, I'll update this to use the bank keeper. I'm going to make a TODO for updating the deposit msg server to go through the bank keeper as well for minting the L2 ETH. The only reason I ended up using the mint keeper for burning was because there was already a precedent for minting from there. It should make test setup a lot easier too since we'll now only need to mock one keeper

Reply:
Forgot I was already using bank keeper here, that was a last minute change before I made a PR :sweat_smile: I still think it's worth looking into minting from the bank keeper too, I can make a task for that